stack.go

```diff
@@ -111,6 +113,30 @@ func (c Call) Format(s fmt.State, verb rune) {
 		buf := [6]byte{}
 		s.Write(strconv.AppendInt(buf[:0], int64(line), 10))
 
+	case 'k':
+		pkg := c.fn.Name()
+
+		var fullPath string
+		var start, end int
+
+		const pathSep = "/"
+		const pkgSep = "."
+		if i := strings.LastIndex(pkg, pathSep); i != -1 {
+			start = (i + len(pathSep))
+			if end = strings.Index(pkg[i+len(pathSep):], pkgSep); end != -1 {
+				end += (i + len(pkgSep))
+				fullPath = pkg[:end]
+			}
+		}
+
+		if s.Flag('+') {
+			start = 0
+		}
+
+		pkg = fullPath[start:end]
```

Comment:
It looks like `fullPath` won't get set if either of the first two `if` statements are not taken, which would result in writing an empty string.

Here is my attempt:
```go
case 'k':
	name := c.fn.Name()
	const pathSep = "/"
	start, end := 0, len(name)
	if i := strings.LastIndex(name, pathSep); i != -1 {
		start = i + len(pathSep)
	}
	const pkgSep = "."
	if i := strings.Index(name[start:], pkgSep); i != -1 {
		end = start + i
	}
	if s.Flag('+') {
		start = 0
	}
	io.WriteString(s, name[start:end])
```
What do you think?